frb_codegen/src/commands.rs

```diff
@@ -18,11 +17,34 @@ fn call_shell(cmd: &str) -> Output {
     execute_command("sh", &["-c", cmd], None)
 }
 
-pub fn ensure_tools_available() -> Result {
-    let output = call_shell("dart pub global list");
-    let output = String::from_utf8_lossy(&output.stdout);
-    if !output.contains("ffigen") {
-        return Err(Error::MissingExe(String::from("ffigen")));
+pub fn ensure_tools_available(dart_root: &str) -> Result {
+    let toolchain = guess_toolchain(dart_root).unwrap();
+    match toolchain {
+        DartToolchain::Dart if !call_shell("dart --version").status.success() => {
+            return Err(Error::MissingExe("dart".to_string()));
+        }
+        DartToolchain::Flutter if !call_shell("flutter --version").status.success() => {
+            return Err(Error::MissingExe("flutter".to_string()));
+        }
+        _ => {}
```

Comment:
nit: Btw seems that `toolchain` only has "Dart" and "Flutter" variants, so "_" is not needed

Reply:
done in f6b63663ba3970de0a559acf43bd6cecb336d1cc